app/Actions/ProcessPlausibleUrlAction.php

```diff
@@ -0,0 +1,173 @@
+<?php
+
+declare(strict_types=1);
+
+namespace App\Actions;
+
+use App\Models\Achievement;
+use App\Models\Game;
+use App\Models\GameSet;
+use App\Models\System;
+
+/**
+ * For Plausible page views, we want to aggregate like-pages into single groups.
+ * For example, /game/1 and /game/5 should not be considered as separate pages for
+ * the purpose of page view tracking.
+ *
+ * Routes are defined using simple configuration objects that specify their type:
+ * - 'id': Routes that just use a numeric ID (eg: /ticket/123)
+ * - 'string': Routes that use a string parameter (eg: /user/username)
+ * - 'model': Routes that can be accessed by ID or slug (eg: /game/sonic-3-123)
+ * - 'nested': Routes with multiple ID-slug segments (eg: /forums/1-community/16-chit-chat/create-topic)
+ */
+class ProcessPlausibleUrlAction
+{
+    private array $routes = [];
+
+    /**
+     * If you're adding new routes to be tracked, you only need to worry about
+     * updating this function.
+     */
+    public function __construct()
+    {
+        // Routes that support self-healing URLs (or will in the future) with ID lookups.
+        $this->addModelRoute('game', Game::class, 'title');
+        $this->addModelRoute('achievement', Achievement::class, 'title');
+        $this->addModelRoute('hub', GameSet::class, 'title');
+        $this->addModelRoute('system', System::class, 'name');
+
+        // Routes that just use a string parameter.
+        $this->addStringRoute('user', 'username');
+
+        // Routes that just use an ID.
+        $this->addIdRoute('ticket');
+
+        // Routes with nested ID-slug segments.
+        // TODO $this->addNestedRoute('forums', ['category', 'forum']);
+    }
```

Comment:
I tried to make this as simple as possible for adding new routes. Nested routes are not yet implemented, but some TODOs are in place.

Ideally a developer should not need to write regex to add model-bound routes. It should be a fairly trivial thing.